src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/EventSource.cs

```diff
@@ -1253,7 +1253,9 @@ internal unsafe void SetMetadata(byte* pointer, int size, int reserved)
             internal ulong m_Ptr;
             internal int m_Size;
 #pragma warning disable 0649
-            internal int m_Reserved;       // Used to pad the size to match the Win32 API
+#pragma warning disable CA1805
+            internal int m_Reserved = 0;       // Used to pad the size to match the Win32 API
```

Comment:
What's the purpose of this?  I see the comments in the other issue, but this won't be meaningful for stackalloc.  This getting executed depends on the new C# 10 feature for being able to add parameterless public ctors to structs, but such ctors aren't going to be executed as part of a stackalloc; either the stackalloc is going to zero everything out anyway, or if the developer has applied SkipLocalsInit, then the stackalloc won't zero, regardless of this initializer.

Reply:
Yeah, I opened the PR prematurely to try and address the comment, but your points are why I pulled it back to draft mode. In its current form, this is only metadata that won't fix the identified scenario.

Do we have a mechanism for ensuring this value is `0` without adding a public method for init-ing the struct or making the property public? Those two options seem to be the only way to reliably get this value to be 0 if someone sets SkipLocalsInit. At this point, I'm left wondering how frequently these config flags are even set in the wild and if this change is worth attempting in any capacity. We've gone a _long_ while without any attempt at zeroing this field. I'm inclined to close the PR.

Reply:
> Do we have a mechanism for ensuring this value is 0 without adding a public method for init-ing the struct or making the property public?

Even with SkipLocalsInit, it'll be zero'd if someone does:
```C#
descrs[0] = new EventData { DataPointer = ..., Size = ... };
```
rather than:
```C#
descrs[0].DataPointer = ...;
descrs[0].Size = ...;
```
so a mechanism already exists. It only won't be zero'd if a stackalloc was used for the space, SkipLocalsInit was set, and the developer then directly set the public fields.

So if code outside of corelib is using SkipLocalsInit and is stackalloc'ing these EventDatas, it should use the former scheme for filling in the values rather than the latter.

Reply:
This is a purely docs related change then. I'll close the PR. I went looking for docs on the subject but could find any for this use case of `stackalloc` + `new`. Based on your usage and phrasing, I'm assuming setting an element of a `stackalloc`'d array using `new` will use the local memory instead of ending up on the heap, correct?

Reply:
> I'm assuming setting an element of a stackalloc'd array using new will use the local memory instead of ending up on the heap, correct?

Yes, or, rather, it'll use stack memory for the local.  `EventData` is a struct.  `descrs[0] = new EventData { DataPointer = ..., Size = ... };` is equivalent to:
```C#
EventData data = default; // zero's the whole struct
data.DataPointer = ...;
data.Size = ...;
descrs[0] = data;
```
You could similarly do:
```C#
descrs[0] = default; // zero's the whole entry
descrs[0].DataPointer = ...;
descrs[0].Size = ...;
```

Reply:
Amazing, thanks @stephentoub!